java/vector/src/main/java/org/apache/arrow/vector/complex/ListVector.java

```diff
@@ -105,6 +106,23 @@ public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers
   }
 
   @Override
+  public void loadFieldBuffers(BuffersIterator buffersIterator, ArrowBuf buf) {
+    buffersIterator.next();
+    ArrowBuf offsetsData = buf.slice((int) buffersIterator.offset(), (int) buffersIterator.length());
+    offsets.load(offsetsData);
+    if (buffersIterator.length() == 0) {
+      offsets.allocateNew(1);
+      offsets.zeroVector();
+    }
+    lastSet = (int) buffersIterator.length() / 4 - 1;
```

Comment:
so here we guess the number of values based on the buffer size? When do we not know how many values where in the vector?